contracts/epochs/FlowEpoch.cdc

```diff
@@ -460,21 +468,31 @@ pub contract FlowEpoch {
 
     /// Calculates a new token payout for the current epoch
     /// and sets the new payout for the next epoch
-    access(account) fun calculateAndSetRewards(_ newPayout: UFix64?) {
+    access(account) fun calculateAndSetRewards() {
+
+        let stakingAdmin = self.borrowStakingAdmin()
 
-        let rewardsBreakdown = self.borrowStakingAdmin().calculateRewards()
+        // Calculate rewards for the current epoch that is about to end
+        // and save that reward breakdown in the epoch metadata for the current epoch
+        let rewardsBreakdown = stakingAdmin.calculateRewards()
         let currentMetadata = self.getEpochMetadata(self.currentEpochCounter)!
         currentMetadata.setRewardAmounts(rewardsBreakdown)
         self.saveEpochMetadata(currentMetadata)
 
-        // Calculate the new epoch's payout
-        // disabled until we enable automated rewards calculations
-        // let newPayout = FlowToken.totalSupply * (FlowEpoch.configurableMetadata.FLOWsupplyIncreasePercentage / 52.0)
+        if FlowEpoch.automaticRewardsEnabled() {
+            // Calculate the total supply of FLOW after the current epoch's payout
+            // the calculation includes the tokens that haven't been minted for the current epoch yet
+            let currentPayout = FlowIDTableStaking.getEpochTokenPayout()
+            let feeAmount = FlowIDTableStaking.getFeesBalance()
+            let flowTotalSupplyAfterPayout = FlowToken.totalSupply + (currentPayout - feeAmount)
```

Comment:
I don't think this handles the case where fees exceed the payout.

[We burn the excess fees](https://github.com/onflow/flow-core-contracts/blob/master/contracts/FlowIDTableStaking.cdc#L910-L911) in that case, so `flowTotalSupplyAfterPayout` should remain the same value as `FlowToken.totalSupply`.

Reply:
yep, good catch. I'll change that